### DISCUSSION TITLE
Partial search / Search issues by a partial word / Wildcard search

### DISCUSSION THREAD

**Discussion by natalia-bearden** [community/community#15298] [Projects and Issues]
Asking for a friend :)
The original request (2018) that I full-heartedly support (2022) is [here](https://github.community/t/feature-request-partial-search-search-issues-by-a-partial-word/1513).  

Thanks so much for anything that could be done to address it!

**Comment by bartlomiej-dawidow** [comment]
Thank you for posting, here's the full original request to keep things in one place:

The suggested feature is to support an asterisk wildcard or a different form of partial search in the issues tab, for example:

`is:issue is:open sustain*`

which should match sustainable , sustainability  etc.

GitHub provides a fairly elaborate search syntax, as explained here:

[https://help.github.com/articles/understanding-the-search-syntax/ 82](https://help.github.com/articles/understanding-the-search-syntax/)

[https://help.github.com/articles/searching-issues-and-pull-requests/ 28](https://help.github.com/articles/searching-issues-and-pull-requests/)

But it does not support partial searches, as discussed here:

[https://stackoverflow.com/a/38532151 57](https://stackoverflow.com/a/38532151)

Quote from GitHub staff:

*Thanks for getting in touch - search doesn’t currently support partial searches, so in your example, you’re correct you would have to search for “feature”. Though not the same, we do however do some stemming for plurals - so searching for “features” would find “feature”.*

This is much more problematic in languages with grammatical cases, such as German, Polish, French, Ukrainian, and many more. To give you an example the Polish word for function  is funkcja but depending on the context it can be any of:

`funkcja`

`funkcji`

`funkcją`

`funkcję`

(and more)

Or plurals:

`funkcje`

`funkcji`

`funkcjami`

`funkcje`

Without partial searches, you have to go through all of the variations to find the matching issues. This makes the search functionality barely usable in languages other than English.

If we were able to search by funkc , we could find all relevant issues in one go. It is a common functionality in search engines. Optionally this could be done with a wildcard, i.e. `funkc*`

**Reply by allanlaal** [reply]
missing wildcards (and OR?) also makes search difficult in English:

`strict*|static* typ*` vs:
 a combination of {strict, strictly + type, typing, typed} + {static, statically, statical + type, typing, typed}  = 14 searches 

it takes me a lot of time to search for duplicate issues when reporting sth - so much so,  that when I'm in a hurry, i dump these into my own todolist waiting for Someday, which will never come.. hundreds of ideas stuck in pre-checks. Or I could just doublereport them and let the repo devs do the same work instead of working on their opensource project :(

**Comment by nazdir** [comment]
Without something like this, it makes my job a pain every day.  I have a team with 32 repos in an org with thousands.  There seems to be no way whatsoever to just see all the PRs that are open for any of my team's repos.  I can see them all for each repo or all for the entire org.  There are 100 ways to solve this, with wildcards and ORs being two of them.  My team only has four devs.  author:adam|beth|charlie|dave would give me what I need.  All our repos start with the project name.  repo:awesome-project* would give me what I need.

**Comment by ghost** [comment]
Issues like keeps "github issue-tracking" in the "toy" category.

**Comment by jelenicak** [comment]
I'd also like to be do this for labels, i.e. find all issues that have the label like "team-name-*". My organisation has a lot of labels, this would be really handy.

**Comment by hatamiarash7** [comment]
I need this too. Something like this:

```
is:pr author:@me -repo:hatamiarash7/* 
```

I want to see all PRs I opened and also exclude personal ones.

**Comment by Timmeey86** [comment]
Not having wildcards in issue searches will hurt any bigger project, especially in an environment where a larger group of non-native English speakers adds issues in English and smaller typos are common, especially in the word ending. Even just a "word begins with" feature would be a game changer. This is such a crucial feature for me that I currently suggest to people to rather track their issues in a different tool.

**Comment by thompson-shaun** [comment]
Interesting (and a bit jarring) that I can do this in a Project

```
is:issue state:open label:"area/*"   
```

**Reply by jeska** [reply]
Yeah I was just playing around with this today. Really handy in Project searches, for sure, but it would be nice to also have this in regular issue search when filtering by label. This would also be nicely solved by scoped labels: https://github.com/orgs/community/discussions/16682

**Comment by momentmal** [comment]
still not a thing?

**Comment by bartlomiej-dawidow** [comment]
Nope, the corporate discrimination continues.

**Comment by silasthegoldenwizard** [comment]
<p>Adding a concrete <strong>issue-number lookup</strong> use case to this request, with current evidence from 27 September 2026.</p>
<p>When returning to an issue on mobile, I may remember only its last few digits. I would like to filter the repository’s issue list by those digits without matching the same numbers elsewhere in the issue text.</p>
<p>In <code>openai/codex</code>, I tested these queries in the repository’s main Issues search field:</p>

Query | Observed result
-- | --
is:issue 48771 state:open | Finds openai/codex#48771.
is:issue 771 state:open | Shows 34 open results, including #44773 and #47015.
is:issue *771 state:open | Also shows 34 open results, including those issues.


<p>A separate check through <code>GET /search/issues</code>, scoped with <code>repo:openai/codex</code>, returned all 34 results for each partial query (<code>incomplete_results: false</code>, <code>per_page=100</code>). <strong>Issue <code>#48771</code> was absent from both result sets</strong>, although the complete-number query found it. No returned issue number ended in <code>771</code>. This API check used the default search mode; I am not assuming that the web UI and API use identical search modes.</p>
<p>There is a valid text match in at least one result: <a href="https://github.com/openai/codex/issues/44773">openai/codex#44773</a> contains <code>execute.rs:771</code> in its description. This is therefore a request for an explicit number filter, rather than a claim that those text matches are incorrect.</p>
<p><strong>Requested capability:</strong> repository-scoped filtering on the issue’s own number. A small initial scope would be exact and suffix matching, for example:</p>
<ul>
<li><code>number:48771</code> — exact number.</li>
<li><code>number:*771</code> — number ends in <code>771</code>.</li>
</ul>
<p>These are proposed examples, not existing documented syntax. An equivalent UI filter would also meet the need; substring matching such as <code>number:*771*</code> could be a later extension.</p>
<p><strong>Suggested acceptance criteria:</strong></p>
<ol>
<li>A suffix query includes issue <code>#48771</code> when it satisfies the other active filters, even if <code>771</code> is absent from its title, description and comments.</li>
<li>It excludes issue <code>#44773</code> when <code>771</code> occurs only in its text.</li>
<li>Repository, state and label filters continue to apply.</li>
<li>The number filter is discoverable on mobile, documents the matching rules, and reports unsupported patterns clearly.</li>
<li>Existing free-text searches retain their current meaning.</li>
</ol>
<p>This would help users recover an issue from a partially remembered identifier. Restricting the first version to one repository and a documented minimum suffix length would be acceptable if needed for performance.</p>
<p>I can help validate the behavior against concrete inclusion/exclusion cases and test any preview. Is this use case best tracked here, or would a dedicated issue-number filtering discussion help?</p>
<p>References: <a href="https://docs.github.com/en/search-github/searching-on-github/searching-issues-and-pull-requests">documented issue search</a> and <a href="https://github.blog/changelog/2026-04-02-improved-search-for-github-issues-is-now-generally-available/">current search modes</a>.</p>
<p><strong>Screenshots:</strong> complete number (<code>48771</code>), suffix with asterisk (<code>*771</code>), and partial number without asterisk (<code>771</code>), in that order.</p>
Adding a concrete issue-number lookup use case to this request, with current evidence from 27 September 2026.

When returning to an issue on mobile, I may remember only its last few digits. I would like to filter the repository’s issue list by those digits without matching the same numbers elsewhere in the issue text.

In "openai/codex", I tested these queries in the repository’s main Issues search field:

Query| Observed result
"is:issue 48771 state:open"| Finds "[openai/codex#48771](https://github.com/openai/codex/issues/48771)" (https://github.com/openai/codex/issues/48771).
"is:issue 771 state:open"| Shows 34 open results, including "[#44773](https://github.com/openai/codex/issues/44773)" (https://github.com/openai/codex/issues/44773) and "[#47015](https://github.com/openai/codex/issues/47015)" (https://github.com/openai/codex/issues/47015).
"is:issue *771 state:open"| Also shows 34 open results, including those issues.

A separate check through "GET /search/issues", scoped with "repo:openai/codex", returned all 34 results for each partial query ("incomplete_results: false", "per_page=100"). Issue "#48771" was absent from both result sets, although the complete-number query found it. No returned issue number ended in "771". This API check used the default search mode; I am not assuming that the web UI and API use identical search modes.

There is a valid text match in at least one result: "[openai/codex#44773](https://github.com/openai/codex/issues/44773)" (https://github.com/openai/codex/issues/44773) contains "execute.rs:771" in its description. This is therefore a request for an explicit number filter, rather than a claim that those text matches are incorrect.

Requested capability: repository-scoped filtering on the issue’s own number. A small initial scope would be exact and suffix matching, for example:

- "number:48771" — exact number.
- "number:*771" — number ends in "771".

These are proposed examples, not existing documented syntax. An equivalent UI filter would also meet the need; substring matching such as "number:*771*" could be a later extension.

Suggested acceptance criteria:

1. A suffix query includes issue "#48771" when it satisfies the other active filters, even if "771" is absent from its title, description and comments.
2. It excludes issue "#44773" when "771" occurs only in its text.
3. Repository, state and label filters continue to apply.
4. The number filter is discoverable on mobile, documents the matching rules, and reports unsupported patterns clearly.
5. Existing free-text searches retain their current meaning.

This would help users recover an issue from a partially remembered identifier. Restricting the first version to one repository and a documented minimum suffix length would be acceptable if needed for performance.

I can help validate the behavior against concrete inclusion/exclusion cases and test any preview. Is this use case best tracked here, or would a dedicated issue-number filtering discussion help?

References: "[documented issue search](https://docs.github.com/en/search-github/searching-on-github/searching-issues-and-pull-requests)" (https://docs.github.com/en/search-github/searching-on-github/searching-issues-and-pull-requests) and "[current search modes](https://github.blog/changelog/2026-04-02-improved-search-for-github-issues-is-now-generally-available/)" (https://github.blog/changelog/2026-04-02-improved-search-for-github-issues-is-now-generally-available/).

Screenshots: complete number ("48771"), suffix with asterisk ("*771"), and partial number without asterisk ("771"), in that order.
<img width="658" height="608" alt="26290-cropped" src="https://github.com/user-attachments/assets/9344c15e-ccae-4ac1-8f9f-a7cc71898284" />
<img width="658" height="906" alt="26291-cropped" src="https://github.com/user-attachments/assets/08f38961-9367-4fe5-a52c-4db600f6b799" />
<img width="658" height="906" alt="26292-cropped" src="https://github.com/user-attachments/assets/4e1f038f-4c07-4ae6-9da5-8d993c670315" />
